javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector("ul");
+const sortType = document.getElementById("sort-selection");
+//dispaly products
+function renderProducts(products) {
+  // clear list before displaying new list
+  clearList();
+
+  //create list of countries
+  products.forEach((product) => {
+    let shipsTo = "";
+    product.shipsTo.forEach((country) => (shipsTo += `<li>${country}</li>`));
+
+    const li = document.createElement("li");
+    li.className = "prduct-display";
+    li.innerHTML = `<ul>
+    <li>${product.name}</li>
+    <li> ${product.price}</li> 
+    <li> ${product.rating}</li>
+    <li> <ul>${shipsTo}</ul></li>
+    </ul>`;
+    productsUl.appendChild(li);
+  });
+}
+// display whole list first time
+renderProducts(getSorted(products));
+
+//display products matches to names intered by user
+const searchInputField = document.getElementById("serach-input");
+
+searchInputField.addEventListener("keyup", () =>
+  renderProducts(getSorted(getSearchProductByName(products)))
+);
+
+// Filter products on name field
+function getSearchProductByName(products) {
+  const searchInput = searchInputField.value.toLowerCase();
+  const searchedProducts = products.filter((product) =>
+    product.name.toLowerCase().includes(searchInput)
+  );
+  return searchedProducts;
+}
+
+//function to remove all li from ul
+function clearList() {
+  while (productsUl.children.length > 0) {
+    productsUl.removeChild(productsUl.childNodes[0]);
+  }
+}
```

Comment:
Hi Qamar, Why did you use 'function to remove all li from ul' would like know.

Reply:
Thank you Soujanya for review my home work. I wanted to clear old list before displaying new list.